e2e/playground.py

```diff
@@ -1691,95 +1678,65 @@ def __init__(self, page: Page, id: str) -> None:
-        self.date_end.set(value=value[1], timeout=timeout)
+        start = value[0]
+        end = value[1]
+        # TODO-future; Composable set() methods?
+        if start is not None:
+            self.date_start.set(value=start, timeout=timeout)
+        if end is not None:
+            self.date_end.set(value=end, timeout=timeout)
 
     def expect_value(
         self,
-        # TODO-barret; Should this be a list or a tuple?
-        value: typing.Union[
-            typing.Tuple[
-                typing.Union[datetime.date, PatternOrStr, None],
-                typing.Union[datetime.date, PatternOrStr, None],
-            ],
-            None,
+        value: typing.Tuple[
+            AttrValue,
+            AttrValue,
         ],
         *,
         timeout: Timeout = None,
     ) -> None:
 
-        start_val = None
-        end_val = None
-        if not (value is None):
-            # Not using `_date_str()` as we want ability to supply non ISO formatted dates
-            start_val = value[0]
-            if isinstance(start_val, datetime.date):
-                start_val = str(start_val)
-            end_val = value[1]
-            if isinstance(end_val, datetime.date):
-                end_val = str(end_val)
+        start_val = value[0]
+        end_val = value[1]
 
-        if start_val is None:
-            start_val = ""
-        if end_val is None:
-            end_val = ""
+        if start_val is None and end_val is None:
+            raise ValueError("Both `start_val` and `end_val` can not be `None`")
 
         # We can not use `[value={value}]` within Locators.
         # The physical `value` attribute is never set, so we can not select on it.
         # We must as the start and end values individually, rather than at the same time like the checkboxgroup input.
-        self.date_start.expect_value(start_val, timeout=timeout)
-        self.date_end.expect_value(end_val, timeout=timeout)
-
-        # loc_dates = self.loc_container
-        # loc_dates = loc_dates.locator(
-        #     "xpath=.",
-        #     has=self.page.locator(f"input:nth-child(1)[type=text][value={start_val}]"),
-        # )
-        # loc_dates = loc_dates.locator(
-        #     "xpath=.",
-        #     has=self.page.locator(f"input:nth-child(3)[type=text][value={end_val}]"),
-        # )
-        # playwright_expect(loc_dates).to_have_count(
-        #     int(not (value is None)),
-        #     timeout=timeout,
-        # )
-        # if False:
-        #     # TODO-future?; This should be in a try catch around the item above for better diagnostics
-        #     self.date_start.expect_value(start_val, timeout=timeout)
-        #     self.date_end.expect_value(end_val, timeout=timeout)
+        # TODO-future; Composable expectations
+        if not (start_val is None):
+            self.date_start.expect_value(start_val, timeout=timeout)
+        if not (end_val is None):
+            self.date_end.expect_value(end_val, timeout=timeout)
```

Comment:
Let's revisit the inconsistency between what None means here versus in set_text.